sklearn/utils/estimator_checks.py

```diff
@@ -1136,13 +1168,20 @@ def check_sample_weight_equivalence(name, estimator_orig):
     y_weighted = _enforce_estimator_tags_y(estimator_weighted, y_weighted)
     y_repeated = _enforce_estimator_tags_y(estimator_repeated, y_repeated)
 
-    estimator_repeated.fit(X_repeated, y=y_repeated, sample_weight=None)
-    estimator_weighted.fit(X_weigthed, y=y_weighted, sample_weight=sw)
+    with warnings.catch_warnings(record=True):
+        # Ensure we converge, otherwise debugging sample_weight equivalence
+        # failures can be very misleading.
+        warnings.simplefilter("error", category=ConvergenceWarning)
+
+        estimator_repeated.fit(X_repeated, y=y_repeated, sample_weight=None)
+        estimator_weighted.fit(X_weigthed, y=y_weighted, sample_weight=sw)
```

Comment:
Why should we ensure convergence ? In my understanding, the deterministic fit should yield the same results for the weighted/repeated data at each iteration, and we can check equivalence after a small number of iterations even if the fit is terrible and hasn't converged. This should ensure that the computation before the loop and in the loop are sample weight aware, but arguably this doesn't test that the stopping criteria is sample weight aware.

I think having many iterations increases the risk to accumulate numerical errors and have false positives.

Reply:
I think the point is to make sure that:

- we do not raise uncaught warning when running the common tests: this forces us to specify estimator parameters either with high `tol` or a large `max_iter`;
- to make the tests, more sensitive, we might want to specify low `tol` values so that the sample weight influence has been full imprinted in the learned decision function. But we want to make sure that `max_iter` is also high enough to make sure that the choice of `tol` has the effect we intend.